src/hooks/use-subreddit-form/use-subreddit-form.js

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+const useSubredditForm = (initialSubreddit, history) => {
+  const [subreddit, setSubreddit] = useState(initialSubreddit);
+
+  const changeSubreddit = (e) => setSubreddit(e.target.value);
+
+  const searchSubreddit = (e) => {
```

Comment:
Great that you detected the unnecessary useEffect. Makes the state also much simpler.